test/engine/BinaryExportTest.cpp

```diff
@@ -0,0 +1,237 @@
+}
+
+// _____________________________________________________________________________
+TEST(BinaryExportHelpers, readVectorOfStrings) {
+  std::vector<uint8_t> data;
+  std::vector<std::string> expected = {"foo", "bar", "baz"};
+
+  auto writeString = [&data](const std::string& str) {
+    size_t size = str.size();
+    for (size_t i = 0; i < sizeof(size); ++i) {
+      data.push_back(static_cast<uint8_t>((size >> (i * 8)) & 0xFF));
+    }
+    for (char c : str) {
+      data.push_back(static_cast<uint8_t>(c));
+    }
+  };
+
+  // Write strings
+  for (const auto& str : expected) {
+    writeString(str);
+  }
+
+  // Write empty string to terminate
+  writeString("");
+
+  auto it = data.begin();
+  auto end = data.end();
+
+  std::vector<std::string> result =
+      BinaryExportHelpers::readVectorOfStrings(it, end);
+  EXPECT_EQ(result, expected);
+  EXPECT_EQ(it, end);
+}
+
+// _____________________________________________________________________________
+TEST(BinaryExportHelpers, rewriteVocabIds) {
+  auto* qec = ad_utility::testing::getQec();
+
+  // Create a test IdTable with local vocab indices
+  IdTable table{2, ad_utility::makeUnlimitedAllocator<Id>()};
+  table.push_back({Id::makeFromInt(42), Id::makeFromLocalVocabIndex(
+                                            reinterpret_cast<LocalVocabIndex>(
+                                                0 << Id::numDatatypeBits))});
+  table.push_back({Id::makeFromInt(43), Id::makeFromLocalVocabIndex(
+                                            reinterpret_cast<LocalVocabIndex>(
+                                                1 << Id::numDatatypeBits))});
+
+  LocalVocab vocab;
+  std::vector<std::string> transmittedStrings = {"<http://example.org/a>",
+                                                  "\"literal\""};
+
+  // Rewrite vocab IDs starting from index 0
+  BinaryExportHelpers::rewriteVocabIds(table, 0, *qec, vocab,
+                                       transmittedStrings);
+
+  // Check that local vocab indices were rewritten
+  // The first column should remain unchanged (integers)
+  EXPECT_EQ(table(0, 0), Id::makeFromInt(42));
+  EXPECT_EQ(table(1, 0), Id::makeFromInt(43));
+
+  // The second column should have been converted to LocalVocabIndex IDs
+  EXPECT_EQ(table(0, 1).getDatatype(), Datatype::LocalVocabIndex);
+  EXPECT_EQ(table(1, 1).getDatatype(), Datatype::LocalVocabIndex);
+}
+
+// _____________________________________________________________________________
+TEST(BinaryExportHelpers, getPrefixMapping) {
+  auto* qec = ad_utility::testing::getQec();
+
+  std::vector<std::string> remotePrefixes = {"<http://example.org/",
+                                              "<http://other.org/"};
+
+  auto mapping =
+      BinaryExportHelpers::getPrefixMapping(*qec, remotePrefixes);
+
+  // The mapping should be empty or contain mappings only for prefixes
+  // that exist in the local index
+  EXPECT_TRUE(mapping.empty() || mapping.size() <= remotePrefixes.size());
```

Comment:
This unit test does way to little and this assertion for `mapping.empty()` is not required.